### ISSUE TITLE
Waybar Does not persist losing monitor connection

### ISSUE THREAD

**Issue by amiles5** [Alexays/Waybar#2699]
I have Arcolinux Hyprland on a Monitor with 3 inputs - one has Macos, another Windows 11 and a third on Arcoliunux Hyprland. Each is on dedicated hardware. If I switch away from Hyprland and return the waybar has disappeared but the workspaces are still there but disorganised. Waybar has stopped for some reason and can be restarted manually. Looking at the Hyprland logs it appears that Monitor removal is causing the issue.

Basically everything survives the switch apart from waybar. eg terminals web browser etc.

Any Ideas how I can stop this happening? Or maybe it is by design?

**Comment by geoffcorey** [comment]
Seeing the same thing.   I manually restarted and when I kill the monitors I see these types of errors on Hyprland

```

[2023-11-28 09:40:40.674] [warning] module keyboard-state: Disabling module "keyboard-state", Failed to find keyboard device: EACCES Permission denied

(waybar:2609408): GLib-GObject-CRITICAL **: 09:40:40.686: ../glib/gobject/gsignal.c:2088: type 'GtkWindow' is already overridden for signal id '73'

(waybar:2609408): GLib-GObject-CRITICAL **: 09:40:40.686: ../glib/gobject/gsignal.c:2088: type 'GtkWindow' is already overridden for signal id '72'
[2023-11-28 09:40:40.693] [warning] Failed to open bar ipc connection: Socket path is empty
[2023-11-28 09:40:40.697] [warning] No batteries.
[2023-11-28 09:40:40.697] [warning] module backlight: Disabling module "backlight", No backlight found
[2023-11-28 09:40:40.698] [warning] Can't open /dev/input/event18 (are you in the input group?): EACCES Permission denied
[2023-11-28 09:40:40.698] [warning] Can't open /dev/input/mouse2 (are you in the input group?): EACCES Permission denied
[2023-11-28 09:40:40.698] [warning] Can't open /dev/input/event17 (are you in the input group?): EACCES Permission denied
[2023-11-28 09:40:40.698] [warning] Can't open /dev/input/mouse1 (are you in the input group?): EACCES Permission denied
```

**Comment by geoffcorey** [comment]
Follow up: 

Made this change to the udev rules but waybar is not crashing anymore.  the idVendor tag is for my USB Bluetooth Dongle found via `lsusb`

```
# cat /etc/udev/rules.d/50-usb_power_save.rules 
ACTION=="add", SUBSYSTEM=="usb", TEST=="power/control", ATTR{power/control}="auto"
ACTION=="add", SUBSYSTEM=="usb", ATTR{idVendor}=="0a12", ATTR{idProduct}=="0001", ATTR{power/autosuspend}="-1"
```

**Comment by hotsno** [comment]
I am also using Hyprland and having the same issue.

@geoffcorey how did you figure out the USB dongle was causing it?

**Comment by geoffcorey** [comment]
Sorry for the delay @hotsno ,   I started the waybar in a terminal to see the output from waybar.
